libcloud/compute/drivers/ecs.py

```diff
@@ -0,0 +1,1390 @@
+
+        @inherits: :class:`NodeDriver.create_node`
+
+        :keyword  ex_node_ids: a list of node's ids used to filter nodes.
+                               Only the nodes which's id in this list
+                               will be returned.
+        :type   ex_node_ids: ``list`` of ``str``
+        :keyword  ex_filters: node attribute and value pairs to filter nodes.
+                              Only the nodes which matchs all the pairs will
+                              be returned.
+                              If the filter attribute need a json array value,
+                              use ``list`` object, the driver will convert it.
+        :type   ex_filters: ``dict``
+        """
+
+        params = {'Action': 'DescribeInstances',
+                  'RegionId': self.region}
+
+        if ex_node_ids:
+            if isinstance(ex_node_ids, list):
+                params['InstanceIds'] = json.dumps(ex_node_ids)
+            else:
+                raise AttributeError('ex_node_ids should be a list of '
+                                     'node ids.')
+
+        if ex_filters:
+            if isinstance(ex_filters, dict):
+                params.update(ex_filters)
+            else:
+                raise AttributeError('ex_filters should be a dict of '
+                                     'node attributes.')
+
+        nodes = self._request_multiple_pages(self.path, params,
+                                             self._to_nodes)
+        return nodes
+
+    def list_sizes(self, location=None):
+        params = {'Action': 'DescribeInstanceTypes'}
+
+        resp_body = self.connection.request(self.path, params).object
+        size_elements = findall(resp_body, 'InstanceTypes/InstanceType',
+                                namespace=self.namespace)
+        sizes = [self._to_size(each) for each in size_elements]
+        return sizes
+
+    def list_locations(self):
+        params = {'Action': 'DescribeRegions'}
+
+        resp_body = self.connection.request(self.path, params).object
+        location_elements = findall(resp_body, 'Regions/Region',
+                                    namespace=self.namespace)
+        locations = [self._to_location(each) for each in location_elements]
+        return locations
+
+    def create_node(self, **kwargs):
+        name = kwargs['name']
+        size = kwargs['size']
+        image = kwargs['image']
+
+        params = {'Action': 'CreateInstance',
+                  'RegionId': self.region,
+                  'ImageId': image.id,
+                  'InstanceType': size.id,
+                  'InstanceName': name}
+
+        if 'ex_security_group_id' not in kwargs:
+            raise AttributeError('ex_security_group_id is mandatory')
+        params['SecurityGroupId'] = kwargs['ex_security_group_id']
+
+        if 'ex_description' in kwargs:
+            params['Description'] = kwargs['ex_description']
+
+        if 'ex_internet_charge_type' in kwargs:
```

Comment:
Maybe this could be refactored in a utility method.

In any case, not a big deal.